systems/framework/lcm_driven_loop.h

```diff
@@ -4,90 +4,207 @@
-/// the user has to provide one extra lcm message (we call it switch message)
-/// which tells LcmDrivenLoop the channel that it should listen to for the
-/// simulation update.
+/// In the case of multiple message types (we call these messages the input
+/// messages), the user has to provide one extra lcm message type (we call this
+/// message the switch message) which tells LcmDrivenLoop the channel that it
+/// should listen to for the simulation update.
+
+/// WARNING: Currently, we update the value of the first InputPort of
+/// `lcm_parser` with the incoming lcm message. This would cause a problem if
+/// the user has a `lcm_parser` with multiple InputPort's.
+
+/// Procedures to use LcmDrivenLoop:
+/// 1. construct LcmDrivenLoop
+/// 2. (if it's multi-input) the user can set the initial channel that
+///    LcmDrivenLoop listens to by calling SetInitActiveChannel().
+/// 3. run Simulate()
+
+/// Note that we implement the class only in the header file because we don't
+/// know what MessageTypes are beforehand.
 
 // We set a default value for SwitchMessageType so that we can generalize this
-// to both single and multi inputs. (for template instantiation)
+// to both single and multi inputs.
 template <typename InputMessageType,
-    typename SwitchMessageType = dairlib::lcmt_controller_switch>
+          typename SwitchMessageType = dairlib::lcmt_controller_switch>
 class LcmDrivenLoop {
  public:
   DRAKE_NO_COPY_NO_MOVE_NO_ASSIGN(LcmDrivenLoop)
 
-  /// Inputs of the single-input LcmDrivenLoop constructor:
-  ///     drake_lcm: DrakeLcm
-  ///     diagram: a drake diagram
-  ///     first_leafsystem: the first block of the diagram
-  ///     input_channel: the name of the input channel
-  ///
-  /// Inputs of the multi-input LcmDrivenLoop constructor:
-  ///     drake_lcm: DrakeLcm
-  ///     diagram: a drake diagram
-  ///     first_leafsystem: the first block of the diagram
-  ///     switch_channel: the name of the switch channel
-  ///     input_channels: the names of the input channels
-
-  /// Procedures to use LcmDrivenLoop:
-  /// 1. construct LcmDrivenLoop
-  /// 2. (if it's multi-input) the user can set the initial channel that
-  ///    LcmDrivenLoop listens to by calling SetInitActiveChannel().
-  /// 3. run Simulate()
-
-  // Lcm driven loop for single input
+  /// Constructor for single-input LcmDrivenLoop
+  ///     @param drake_lcm DrakeLcm
+  ///     @param diagram A Drake diagram
+  ///     @param lcm_parser The LeafSystem of the diagram that parses the
+  ///     incoming lcm message
+  ///     @param input_channel The name of the input channel
   LcmDrivenLoop(drake::lcm::DrakeLcm* drake_lcm,
                 std::unique_ptr<drake::systems::Diagram<double>> diagram,
-                const drake::systems::LeafSystem<double>* first_leafsystem,
-                const std::string& input_channel);
-
-  // Lcm driven loop for multiple inputs
+                const drake::systems::LeafSystem<double>* lcm_parser,
```

Comment:
Can/should this just call the multi-channel constructor (with just a length 1 list)? If so, I don't see a need for the private constructor.

Reply:
Yes, we can have one multi-channel constructor only, but I think it is unnatural to create a length 1 list for the single input channel

Reply:
I will change to single constructor for shorter code anyways.

Reply:
I think having two constructors is good, but I was questioning the need for the first. I think something like:

```
  LcmDrivenLoop(drake::lcm::DrakeLcm* drake_lcm,
                std::unique_ptr<drake::systems::Diagram<double>> diagram,
                const drake::systems::LeafSystem<double>* lcm_parser,
                const std::string& input_channel)
      : LcmDrivenLoop(drake_lcm, std::move(diagram), input_channel, "", std::vector<std::string>{input_channel}, lcm_parser) {}
```

where the single-channel constructor just calls the multi-channel constructor (with a blank switching channel argument and length 1 vector)